### PR TITLE
add  Gitpod doc

### LINKED ISSUE #2429: Add documentation on how to use GitPod to develop/run the code
We recently added support for [GitPod](https://www.gitpod.io/) in #2406.  It's not fully ready yet (see #2427), but it's close, and the integration is complete in the repo, see https://github.com/Seneca-CDOT/telescope/issues/2427#issuecomment-955874717.

We need to properly document this, so our devs know what this is, what the implications are of using it, and how to make it all work.  I especially want to make sure people don't run afoul of any pricing issues (you get 50 hours a month for free, which is probably perfect for students fixing a few issues), and know what they are clicking.

cc @DukeManh, @cindyledev.

### PR DESCRIPTION
**Pull request by DukeManh**
Fixes #2429

This adds the Gitpod doc for development on Gitpod.

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [x] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Checklist

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

<a href="https://gitpod.io/#https://github.com/Seneca-CDOT/telescope/pull/2474"></a>

### CODE REVIEW

File: .gitpod.yml [hunk 1]
```diff
@@ -90,7 +90,7 @@ github:
     # add a "Review in Gitpod" button as a comment to pull requests (defaults to false)
     addComment: true
     # add a "Review in Gitpod" button to the pull request's description (defaults to false)
-    addBadge: true
+    addBadge: false
```
Comment (DukeManh) [inline .gitpod.yml:93]: We already have a badge in Gitpod bot comment.

File: docs/gitpod.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+2. Open in Gitpod:
+
+- Option 1: [Install the Gitpod extension](https://www.gitpod.io/docs/browser-extension) to open the page in Gitpod with a click of a button.
+- Option 2: In the address bar, prefix the entire URL with `gitpod.io/#` and press Enter.
+
+3. Sign in with one of the listed providers.
+
+4. You should see a browser-based VSCode serving code of that branch or PR. If you open
+   an Issue, Gitpod automatically checks out a new git branch for you to work on.
```
Comment (humphd) [inline docs/gitpod.md:31]: Add a screenshot of this

File: docs/gitpod.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+5. On the first tab of the VScode terminal, the left half is for building and starting
+   services, the second half waits for services to start and spins up the Next frontend dev server.
+   The second tab is free for writing commands.
```
Comment (humphd) [inline docs/gitpod.md:35]: Same, add a screenshot.  Perhaps 1 will do, and both 4 and 5 can use it.

File: docs/gitpod.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+## Gitpod pre-builds
+
+Gitpod continuously builds all your Git branches like a CI server. Whenever your code changes
```
Comment (humphd) [inline docs/gitpod.md:39]: The use of "your" is confusing here.  It should be "Telescope's".  You should also mention pull requests.

File: docs/gitpod.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+(e.g. when new commits are pushed to your repository), Gitpod can prebuild workspaces,
+i.e. run the init commands in your .gitpod.yml configuration file before you even start a workspace.
+
+## Shut down your workspaces
+
+Within 3 minutes after closing your workspace browser tab, it will be stopped.
```
Comment (humphd) [inline docs/gitpod.md:45]: Add the note about 50 free hours per month here, so it's clear why this matters.

File: docs/gitpod.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+Unused workspaces are automatically deleted after 14 days of inactivity.
+To prevent a workspace from being deleted, you can pin it in your list of workspaces.
+Pinned workspaces are kept forever.
+
+## Resuming workspaces
+
+Navigate to [Gitpod Dashboard](https://gitpod.io/workspaces) to select a workspace to continue
+working on.
+
+## Gitpod Pricing
```
Comment (humphd) [inline docs/gitpod.md:56]: Move this up

**Review by humphd — CHANGES_REQUESTED** [review]
This is excellent to have

File: docs/CONTRIBUTING.md [hunk 7]
```diff
@@ -14,12 +14,20 @@ Before creating an issue:
...
 Telescope has many parts, and setup requires you to install a number of tools
 and dependencies. For instructions on how to setup your Telescope environment, please see
 the [Environment Setup documentation](https://github.com/Seneca-CDOT/telescope/blob/master/docs/environment-setup.md).
 
+## Browser-based Setup
+
+If you are just getting started, we recommend setting up Telescope locally first
+to familiarize yourself with the workflows of a typical project.
+
+Besides, you can also start an automated development environment and access it
```
[the remark this answers is not in the document]
Comment (DukeManh) [reply docs/CONTRIBUTING.md:28]: :%s/Besides/However/

**Comment by DukeManh** [comment]
@humphd Can you re-review please. cc @cindyledev.

**Review by humphd — APPROVED** [review]
Amazing.  We should find a student in the class who wants to develop something on Telescope to use this, and test it out to see if it's missing anything or unclear.

**Review by HyperTHD — APPROVED** [review]

**Pull request merged**